src/Features/CSharp/Portable/CodeLens/CSharpCodeLensDisplayInfoService.cs

```diff
@@ -0,0 +1,202 @@
+    {
+        private static readonly SymbolDisplayFormat DefaultFormat =
+            SymbolDisplayFormat.CSharpErrorMessageFormat.RemoveMemberOptions(
+                SymbolDisplayMemberOptions.IncludeExplicitInterface);
+
+        // Matches default ToDisplayString except removing global namspace and namespaces
+        private static readonly SymbolDisplayFormat ShortFormat =
+            DefaultFormat.WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Omitted)
+                .WithTypeQualificationStyle(SymbolDisplayTypeQualificationStyle.NameAndContainingTypes);
+
+        /// <summary>
+        /// Returns the node that should be displayed
+        /// </summary>
+        public SyntaxNode GetDisplayNode(SyntaxNode node)
+        {
+            while (true)
+            {
+                switch (node.Kind())
+                {
+                    // LocalDeclarations do not have symbols themselves, you need a variable declarator
+                    case SyntaxKind.LocalDeclarationStatement:
+                        var localDeclarationNode = (LocalDeclarationStatementSyntax) node;
+                        node = localDeclarationNode.Declaration.Variables.First();
+                        continue;
+
+                        // Field and event declarations do not have symbols themselves, you need a variable declarator
+                    case SyntaxKind.FieldDeclaration:
+                    case SyntaxKind.EventFieldDeclaration:
+                        var fieldNode = (BaseFieldDeclarationSyntax) node;
+                        node = fieldNode.Declaration.Variables.First();
+                        continue;
+
+                        // Variable is a field without access modifier. Parent is FieldDeclaration
+                    case SyntaxKind.VariableDeclaration:
+                        node = node.Parent;
+                        continue;
+
+                        // Built in types   
+                    case SyntaxKind.PredefinedType:
+                        node = node.Parent;
+                        continue;
+
+                    case SyntaxKind.MultiLineDocumentationCommentTrivia:
+                    case SyntaxKind.SingleLineDocumentationCommentTrivia:
+                        // For DocumentationCommentTrivia node, node.Parent is null. Obtain parent through ParentTrivia.Token
+                        if (node.IsStructuredTrivia)
+                        {
+                            var structuredTriviaSyntax = (StructuredTriviaSyntax) node;
+                            node = structuredTriviaSyntax.ParentTrivia.Token.Parent;
+                            continue;
+                        }
+                        return null;
+
+                    default:
+                        return node;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the DisplayName for the given node.
```

Comment:
Yes. When you click on a CodeLens reference indicator, you see a list of references. If you hover over one of those references, you see a tooltip come up that contains information about the reference, including the "name" of the thing that is doing the referencing. That's what this is used for.

---
In reply to: [74127198](https://github.com/dotnet/roslyn/pull/13037#discussion_r74127198) [](ancestors = 74127198)